calc-payout/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+mod debug;
+
+use core::str::FromStr;
+use log::info;
+use sp_arithmetic::Perbill;
+use wasm_bindgen::prelude::*;
+
+type Balance = u128;
+type RewardPoint = u32;
+
+#[wasm_bindgen]
+#[derive(Debug)]
+pub struct CalcPayout {
+    total_reward_points: RewardPoint,
+}
+
+#[wasm_bindgen]
+impl CalcPayout {
+    pub fn from_params(total_reward_points: &str) -> CalcPayout {
+        debug::setup();
+
+        info!("from_params({}) -> ", total_reward_points);
+
+        let calc_payout = Self {
+            total_reward_points: RewardPoint::from_str(total_reward_points).unwrap(),
```

Comment:
You can pass a `u32` (any integer below `u128`)  natively without going through a `&str` or any serialisation using `JsValue`. You already have done that in `calc_payout`.